cli/src/git_util.rs

```diff
@@ -456,47 +457,49 @@ export or their "parent" bookmarks."#,
     Ok(())
 }
 
+// TODO: Move this back to cli/src/commands/git/fetch.rs
+// With the new `GitFetch` api, this function is too specialized
+// to the `jj git fetch` command and should not be reused.
 pub fn git_fetch(
     ui: &mut Ui,
     tx: &mut WorkspaceCommandTransaction,
     git_repo: &git2::Repository,
     remotes: &[String],
     branch: &[StringPattern],
 ) -> Result<(), CommandError> {
-    let git_settings = tx.settings().git_settings();
-
-    for remote in remotes {
-        let stats = with_remote_git_callbacks(ui, None, |cb| {
-            git::fetch(
+    let import_stats =
+        with_remote_git_callbacks(ui, None, |cb| -> Result<GitImportStats, CommandError> {
```

Comment:
Good call. The callback is only initialized once when we cast from RemoteCallback into git2::RemoteCallback. I've gone ahead to rework this so that each `.fetch()` takes a callback and depth, so it doesn't break the progress indicator or any other thing. Shame though, as apart from the progress, other functionality would have been fine.